source/tests/system/nirfmxbt_driver_api_tests.cpp

```diff
@@ -258,14 +244,34 @@ TEST_F(NiRFmxBTDriverApiTests, TxpBasicFromExample_DataLooksReasonable)
-TEST_F(NiRFmxBTDriverApiTests, SetAttributeComplex_ExpectedError)
+TEST_F(NiRFmxBTDriverApiTests, ExerciseComplexNumberUsageWithModAccFetchConstellationTrace_DataLooksReasonable)
 {
   const auto session = init_session(stub(), kPxi5663e);
+  EXPECT_SUCCESS(session, client::cfg_frequency_reference(stub(), session, "", FrequencyReferenceSource::FREQUENCY_REFERENCE_SOURCE_ONBOARD_CLOCK, 10e6));
+  EXPECT_SUCCESS(session, client::cfg_rf(stub(), session, "", 2.402000e9, 0.0, 0.0));
+  EXPECT_SUCCESS(session, client::cfg_iq_power_edge_trigger(stub(), session, "", "0", IQPowerEdgeTriggerSlope::IQ_POWER_EDGE_TRIGGER_SLOPE_RISING_SLOPE, -20.0, 0.0, TriggerMinimumQuietTimeMode::TRIGGER_MINIMUM_QUIET_TIME_MODE_AUTO, 100e-6, IQPowerEdgeTriggerLevelType::IQ_POWER_EDGE_TRIGGER_LEVEL_TYPE_RELATIVE, true));
+  EXPECT_SUCCESS(session, client::cfg_packet_type(stub(), session, "", PacketType::PACKET_TYPE_2_DH1));
+  EXPECT_SUCCESS(session, client::cfg_payload_length(stub(), session, "", PayloadLengthMode::PAYLOAD_LENGTH_MODE_AUTO, 10));
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_MODACC, true));
+  EXPECT_SUCCESS(session, client::mod_acc_cfg_burst_synchronization_type(stub(), session, "", ModAccBurstSynchronizationType::MODACC_BURST_SYNCHRONIZATION_TYPE_PREAMBLE));
+  EXPECT_SUCCESS(session, client::mod_acc_cfg_averaging(stub(), session, "", ModAccAveragingEnabled::MODACC_AVERAGING_ENABLED_FALSE, 10));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
 
-  EXPECT_RFMX_ERROR(
-      -380231, "This attribute is read-only and cannot be written", session,
-      client::set_attribute_ni_complex_double_array(stub(), session, "", NiRFmxBTAttribute::NIRFMXBT_ATTRIBUTE_ACP_RESULTS_REFERENCE_CHANNEL_POWER, complex_number_array({1.2, 2.2}, {1e6, 1.01e6})));
+  // We expect these actions to produce kPreambleSyncPacketStartDetectionFailedWarning since the test uses simulated hardware.
+  const auto devm_response = client::mod_acc_fetch_devm(stub(), session, "", 10.0);
+  EXPECT_WARNING(devm_response, kPreambleSyncPacketStartDetectionFailedWarning);
+  const auto frequency_error_edr_response = client::mod_acc_fetch_frequency_error_edr(stub(), session, "", 10.0);
+  EXPECT_WARNING(frequency_error_edr_response, kPreambleSyncPacketStartDetectionFailedWarning);
+  const auto devm_per_symbol_trace_response = client::mod_acc_fetch_devm_per_symbol_trace(stub(), session, "", 10.0);
+  EXPECT_WARNING(devm_per_symbol_trace_response, kPreambleSyncPacketStartDetectionFailedWarning);
+  const auto frequency_error_wi_plus_w0_trace_edr_response = client::mod_acc_fetch_frequency_error_wi_plus_w0_trace_edr(stub(), session, "", 10.0);
+  EXPECT_WARNING(frequency_error_wi_plus_w0_trace_edr_response, kPreambleSyncPacketStartDetectionFailedWarning);
+  const auto constellation_trace_response = client::mod_acc_fetch_constellation_trace(stub(), session, "", 10.0);
```

Comment:
Other than checking constellation trace are any of the other fetches providing value here?

Reply:
No, I had just copied the LabVIEW example entirely. But yeah, the other fetches provide no value, so I've removed them.